src/app/service.ts

```diff
@@ -24,25 +25,53 @@ export const makeApp = ({
 
     await httpHandler.run()
 
-    const messages = await messagesProcessor.load()
-    const verifiedMessages = await messagesProcessor.verify(messages)
+    /**
+     * Appends new messages to messageStorage,
+     * Not modifying the messagesStorage reference
+     *
+     * Notes:
+     *   - new messages are identified by difference of signature
+     */
+    const reloadAndVerifyMessages = async (messagesStorage: MessageStorage) => {
+      logger.info(`Will reload messages`)
+      logger.info(`Existing messages count ${messagesStorage.length}`)
+      const messages = await messagesProcessor.load()
+      const verifiedMessages = await messagesProcessor.verify(messages)
+
+      logger.info(`Existing messages count ${messagesStorage.length}`)
+
+      const appendedMessagesCount =
+        messagesStorage.addMessages(verifiedMessages)
```

Comment:
Signatures can either be added or deleted. I think we should update the whole list in memory on each cycle. But have a cache for validation and decryption to do it without overloading the processor